src/Engine/ProtoCore/Parser/atg/Associative.atg

```diff
@@ -1055,14 +1055,30 @@ Associative_DecoratedIdentifier<out ProtoCore.AST.AssociativeAST.AssociativeNode
                         .)
         ':'
 
-        ident
                         (. 
-                            int type = core.TypeSystem.GetType(t.val); 
+                            string ident = string.Empty;
+							int type = ProtoCore.DSASM.Constants.kInvalidIndex;
+							if (la.kind == 1)
+							{
+								if (IsIdentList())
+								{
+									Associative_IdentifierList(out node);
```

Comment:
There is a defect in Codegen where a type mismatch warning is thrown with a typed identifier. I have logged it here: http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6604. As far as parsing is concerned there are no parser related errors anymore.